runtime/jvmti/jvmtiStartup.c

```diff
@@ -1061,6 +1066,171 @@ createXrunLibraries(J9JavaVM * vm)
+				|| (0 == strcmp(fileName, libNameAr))
+			) {
+				result = TRUE;
+			}
+		} else {
+			if (!isLibraryPath) {
+				/* libName has no path separator, compare the whole file names */
+				if (0 == strcmp(fileName, libName)) {
+					result = TRUE;
+				}
+			}
+			/* if libName contains a path separator, it can't be the same library as this fileName */
+		}
+	} else {
+		/* find the library name of this object file */
+		const char *tmpLibName = tmp + 1;
+		if (decorate) {
+			/* compare with libName added the prefix/suffix */
+			if ((0 == strcmp(tmpLibName, libNameSo))
+				|| (0 == strcmp(tmpLibName, libNameAr))
+			) {
+				/* check if this object file path is part of libpath */
+				UDATA dirLen = tmp - fileName;
+				result = isFileNameInLibPath(fileName, dirLen, libpath);
+			}
+		} else {
+			/* check if libName contains a path separator */
+			if (isLibraryPath) {
+				/* compare the library directly with the object file loaded */
+				if (0 == strcmp(fileName, libName)) {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_found(fileName);
+					result = TRUE;
+				} else {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_not_found(fileName, libName);
+				}
+			} else {
+				/* libName has no path separator, compare the whole file names */
+				if (0 == strcmp(tmpLibName, libName)) {
+					/* check if this object file path is part of libpath */
+					UDATA dirLen = tmp - fileName;
+					result = isFileNameInLibPath(fileName, dirLen, libpath);
+				}
+			}
+		}
+	}
+	return result;
+}
+
+/**
+ * A loadquery() helper method to reallocate the buffer if the initial allocation is insufficient.
+ * The caller is responsible for freeing the buffer allocated.
+ *
+ * @param[in] vm Java VM
+ * @param[in] flag specifies the action of the loadquery
+ * @param[in/out] buffer points to a buffer in which to store the information
+ * @param[in/out] bufferLength specifies the initial allocation size and returns the actual size of the buffer
+ *
+ * @return TRUE if loadquery() completes successfully, FALSE otherwise
+ */
+static BOOLEAN
+loadqueryHelper(J9JavaVM *vm, int flag, void **buffer, UDATA *bufferLength)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	BOOLEAN result = FALSE;
+	void *tmpBuffer = NULL;
+	UDATA tmpBufferLength = *bufferLength;
+
+	tmpBuffer = j9mem_allocate_memory(tmpBufferLength, J9MEM_CATEGORY_JVMTI);
+	if (NULL != tmpBuffer) {
+		for (;;) {
+			int rc = loadquery(flag, tmpBuffer, tmpBufferLength);
+			if (-1 != rc) {
+				/* loadquery() completed successfully */
+				*buffer = tmpBuffer;
+				*bufferLength = tmpBufferLength;
+				result = TRUE;
+				break;
+			}
+			if (ENOMEM == errno) {
+				void *newBuffer = NULL;
+				/* insufficient buffer size, increase it */
+				tmpBufferLength *= 2;
+				newBuffer = j9mem_reallocate_memory(tmpBuffer, tmpBufferLength, J9MEM_CATEGORY_JVMTI);
+				if (NULL == newBuffer) {
+					j9mem_free_memory(tmpBuffer);
+					Trc_JVMTI_loadqueryHelper_OOM(tmpBufferLength);
```

Comment:
A call to `setNativeOutOfMemoryError()` is missing for this failure mode.

Reply:
Fixed